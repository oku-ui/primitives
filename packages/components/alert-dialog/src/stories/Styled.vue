<script setup lang="ts">
import {
  OkuAlertDialog, OkuAlertDialogAction, OkuAlertDialogCancel, OkuAlertDialogContent, OkuAlertDialogDescription, OkuAlertDialogOverlay, OkuAlertDialogPortal,
  OkuAlertDialogTitle, OkuAlertDialogTrigger,
} from '@oku-ui/alert-dialog'
</script>

<template>
  <div>
    <OkuAlertDialog>
      <OkuAlertDialogTrigger class="alert_dialog_triggerClass">
        delete everything
      </OkuAlertDialogTrigger>
      <OkuAlertDialogPortal>
        <OkuAlertDialogOverlay class="alert_dialog_overlayClass" />
        <OkuAlertDialogContent class="alert_dialog_contentClass">
          <OkuAlertDialogTitle class="alert_dialog_titleClass">
            Are you sure?
          </OkuAlertDialogTitle>
          <OkuAlertDialogDescription class="alert_dialog_descriptionClass">
            This will do a very dangerous thing. Thar be dragons!
          </OkuAlertDialogDescription>
          <OkuAlertDialogAction class="alert_dialog_actionClass">
            yolo, do it
          </OkuAlertDialogAction>
          <OkuAlertDialogCancel class="alert_dialog_cancelClass">
            maybe not
          </OkuAlertDialogCancel>
        </OkuAlertDialogContent>
      </OkuAlertDialogPortal>
    </OkuAlertDialog>
  </div>
</template>
