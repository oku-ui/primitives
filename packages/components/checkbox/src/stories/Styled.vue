<script setup lang="ts">
import { OkuCheckbox, OkuCheckboxIndicator } from '@oku-ui/checkbox'
import Label from './Label.vue'
</script>

<template>
  <p>This checkbox is nested inside a label. The state is uncontrolled.</p>

  <h1>Custom label</h1>
  <Label>
    Label{{ ' ' }}
    <OkuCheckbox class="checkbox">
      <OkuCheckboxIndicator class="checkbox-indicator" />
    </OkuCheckbox>
  </Label>

  <br>
  <br>

  <h1>Native label</h1>
  <label>
    Label{{ ' ' }}
    <OkuCheckbox class="checkbox">
      <OkuCheckboxIndicator class="checkbox-indicator" />
    </OkuCheckbox>
  </label>

  <h1>Native label + native checkbox</h1>
  <label>
    Label <input type="checkbox">
  </label>

  <h1>Custom label + htmlFor</h1>
  <Label for="one">Label</Label>
  <OkuCheckbox id="one" class="checkbox">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <br>
  <br>

  <h1>Native label + htmlFor</h1>
  <label for="two">Label</label>
  <OkuCheckbox id="two" class="checkbox">
    <OkuCheckboxIndicator class="checkbox-indicator" />
  </OkuCheckbox>

  <h1>Native label + native checkbox</h1>
  <label for="three">Label</label>
  <input id="three" type="checkbox">
</template>
