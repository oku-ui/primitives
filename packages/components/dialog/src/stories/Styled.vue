<script setup lang="ts">
import { OkuDialog, OkuDialogClose, OkuDialogContent, OkuDialogDescription, OkuDialogOverlay, OkuDialogPortal, OkuDialogTitle, OkuDialogTrigger } from '@oku-ui/dialog'
</script>

<template>
  <div>
    <OkuDialog>
      <OkuDialogTrigger class="triggerClass">
        open
      </OkuDialogTrigger>
      <OkuDialogPortal>
        <OkuDialogOverlay class="overlayClass" />
        <OkuDialogContent class="contentDefaultClass">
          <OkuDialogTitle>
            Booking info
          </OkuDialogTitle>
          <OkuDialogDescription>
            Please enter the info for your booking below.
          </OkuDialogDescription>
          <OkuDialogClose class="closeClass">
            close
          </OkuDialogClose>
        </OkuDialogContent>
      </OkuDialogPortal>
    </OkuDialog>
  </div>
</template>
