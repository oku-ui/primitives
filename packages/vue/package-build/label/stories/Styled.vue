<script setup lang="ts">
import { OkuLabel } from '@oku-ui/label'
</script>

<template>
  <OkuLabel class="label">
    Label
  </OkuLabel>
</template>
