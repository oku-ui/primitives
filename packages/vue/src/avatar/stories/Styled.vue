<script setup lang="ts">
import { OkuAvatar, OkuAvatarFallback, OkuAvatarImage } from '@oku-ui/avatar'
import AvatarIcon from './AvatarIcon.vue'

const src = 'https://picsum.photos/id/1005/400/400'
const srcBroken = 'https://broken.link.com/broken-pic.jpg'
</script>

<template>
  <h1>Without image & with fallback</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarFallback class="avatar-fallback">
      JS
    </OkuAvatarFallback>
  </OkuAvatar>

  <h1>With image & with fallback</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarImage class="avatar-image" alt="John Smith" :src="src" />
    <OkuAvatarFallback :delay-ms="300" class="avatar-fallback">
      JS
    </OkuAvatarFallback>
  </OkuAvatar>

  <h1>With image & with fallback (but broken src)</h1>
  <OkuAvatar class="avatar">
    <OkuAvatarImage
      class="avatar-image"
      alt="John Smith"
      :src="srcBroken"
      @loading-status-change="console.log($event)"
    />
    <OkuAvatarFallback class="avatar-fallback">
      <AvatarIcon />
    </OkuAvatarFallback>
  </OkuAvatar>
</template>
