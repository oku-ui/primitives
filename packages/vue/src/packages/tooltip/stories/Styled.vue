<script setup lang="ts">
import {
  OkuTooltip,
  OkuTooltipArrow,
  OkuTooltipContent,
  OkuTooltipPortal,
  OkuTooltipProvider,
  OkuTooltipTrigger,
} from '@oku-ui/tooltip'
</script>

<template>
  <OkuTooltipProvider>
    <OkuTooltip>
      <OkuTooltipTrigger class="tooltip-trigger">
        Hover or Focus me
      </OkuTooltipTrigger>
      <OkuTooltipPortal>
        <OkuTooltipContent class="tooltip-content" :side-offset="5">
          Nicely done!
          <OkuTooltipArrow class="tooltip-arrow" :offset="10" />
        </OkuTooltipContent>
      </OkuTooltipPortal>
    </OkuTooltip>
  </OkuTooltipProvider>
</template>
