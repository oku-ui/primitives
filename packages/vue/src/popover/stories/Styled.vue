<script setup lang="ts">
import {
  OkuPopover,
  OkuPopoverArrow,
  OkuPopoverClose,
  OkuPopoverContent,
  OkuPopoverPortal,
  OkuPopoverTrigger,
} from '@oku-ui/popover'
</script>

<template>
  <div
    :style="{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      height: '200vh',
    }"
  >
    <OkuPopover>
      <OkuPopoverTrigger class="popover-trigger">
        open
      </OkuPopoverTrigger>
      <OkuPopoverPortal>
        <OkuPopoverContent class="popover-content" :side-offset="5">
          <OkuPopoverClose class="popover-close">
            close
          </OkuPopoverClose>
          <OkuPopoverArrow class="popover-arrow" :width="20" :height="10" />
        </OkuPopoverContent>
      </OkuPopoverPortal>
    </OkuPopover>

    <input>
  </div>
</template>
