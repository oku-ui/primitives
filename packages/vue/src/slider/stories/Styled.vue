<script setup lang="ts">
import { OkuSlider, OkuSliderRange, OkuSliderThumb, OkuSliderTrack } from '@oku-ui/slider'
</script>

<template>
  <div>
    <OkuSlider class="slider-rootSlider">
      <OkuSliderTrack class="slider-track">
        <OkuSliderRange class="slider-range" />
      </OkuSliderTrack>
      <OkuSliderThumb class="slider-thumb" />
    </OkuSlider>
  </div>
</template>
