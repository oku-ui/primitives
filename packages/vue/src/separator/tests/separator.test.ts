import { enableAutoUnmount, shallowMount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe'

import { OkuSeparator } from '../'

import Styled from '../stories/Styled.vue'

enableAutoUnmount(afterEach)

describe('okuSeparator', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount(OkuSeparator, {
      attachTo: document.body,
    })
  })

  it('should render OkuSeparator correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()

    expect(shallowMount(OkuSeparator).html()).toMatchSnapshot()
  })

  /**
   * @vitest-environment jsdom
   */
  it('should have no accessibility violations', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })

  it('should render with a horizontal data-orientation attribute by default', () => {
    expect(wrapper.attributes('data-orientation')).toBe('horizontal')
  })

  it('should render with an undefined aria-orientation attribute by default', () => {
    expect(wrapper.attributes('aria-orientation')).toBe(undefined)
  })

  it('should render with a separator role attribute by default', () => {
    expect(wrapper.attributes('role')).toBe('separator')
  })

  it('should log an error when the orientation prop is invalid', () => {
    const spy = vi.spyOn(console, 'error').mockImplementation(vi.fn() as any)

    wrapper = shallowMount(OkuSeparator, {
      props: {
        orientation: 'invalid-orientation' as any,
      },
    })

    expect(spy).toMatchSnapshot()
  })

  describe('given a horizontal Separator', () => {
    beforeEach(() => {
      wrapper = shallowMount(OkuSeparator, {
        props: {
          orientation: 'horizontal',
        },
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render with a data-orientation attribute', () => {
      expect(wrapper.attributes('data-orientation')).toBe('horizontal')
    })

    it('should render with an undefined aria-orientation attribute by default', () => {
      expect(wrapper.attributes('aria-orientation')).toBe(undefined)
    })
  })

  describe('given a vertical Separator', () => {
    beforeEach(() => {
      wrapper = shallowMount(OkuSeparator, {
        props: {
          orientation: 'vertical',
        },
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render with a vertical data-orientation attribute', () => {
      expect(wrapper.attributes('data-orientation')).toBe('vertical')
    })

    it('should render with a vertical aria-orientation attribute by default', () => {
      expect(wrapper.attributes('aria-orientation')).toBe('vertical')
    })
  })

  describe('given a decorative  Separator', () => {
    beforeEach(() => {
      wrapper = shallowMount(OkuSeparator, {
        props: {
          decorative: true,
        },
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render with a none role attribute', () => {
      expect(wrapper.attributes('role')).toBe('none')
    })
  })
})

describe('okuSeparator Stories', () => {
  describe('styled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Styled>>

    beforeEach(async () => {
      wrapper = shallowMount(Styled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
