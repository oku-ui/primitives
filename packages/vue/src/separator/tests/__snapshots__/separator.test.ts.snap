// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuSeparator > should log an error when the orientation prop is invalid 1`] = `
[MockFunction error] {
  "calls": [
    [
      "Invalid prop \`orientation\` of value \`invalid-orientation\` supplied to \`OkuSeparator\`, expected one of:
  - horizontal
  - vertical

Defaulting to \`horizontal\`.",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`okuSeparator > should render OkuSeparator correctly 1`] = `"<oku-primitive-stub is="div" aschild="false" data-orientation="horizontal" role="separator"></oku-primitive-stub>"`;

exports[`okuSeparator > should render OkuSeparator correctly 2`] = `"<oku-primitive-stub is="div" aschild="false" data-orientation="horizontal" role="separator"></oku-primitive-stub>"`;

exports[`okuSeparator Stories > styled > should render correctly 1`] = `
"<h1>Horizontal</h1>
<p>The following separator is horizontal and has semantic meaning.</p>
<oku-separator-stub orientation="horizontal" decorative="false" is="div" aschild="false" class="separator"></oku-separator-stub>
<p> The following separator is horizontal and is purely decorative. Assistive technology will ignore this element. </p>
<oku-separator-stub orientation="horizontal" decorative="true" is="div" aschild="false" class="separator"></oku-separator-stub>
<h1>Vertical</h1>
<div style="display: flex; align-items: center;">
  <p>The following separator is vertical and has semantic meaning.</p>
  <oku-separator-stub orientation="vertical" decorative="false" is="div" aschild="false" class="separator"></oku-separator-stub>
  <p> The following separator is vertical and is purely decorative. Assistive technology will ignore this element. </p>
  <oku-separator-stub orientation="vertical" decorative="true" is="div" aschild="false" class="separator"></oku-separator-stub>
</div>"
`;
